src/libcore/iter/iterator.rs

```diff
@@ -130,9 +130,10 @@ pub trait Iterator {
     ///
     /// ```
     /// // an infinite iterator has no upper bound
+    /// // and the maximum possible lower bound
```

Comment:
Infinite iterators can have any lower bound so I don't think this line should have been added.